lib/loglevel.js

```diff
@@ -34,54 +31,17 @@
     var _loggersByName = {};
     var defaultLogger = null;
 
-    // Cross-browser bind equivalent that works at least back to IE6
-    function bindMethod(obj, methodName) {
-        var method = obj[methodName];
-        if (typeof method.bind === 'function') {
-            return method.bind(obj);
-        } else {
-            try {
-                return Function.prototype.bind.call(method, obj);
-            } catch (e) {
-                // Missing bind shim or IE8 + Modernizr, fallback to wrapping
-                return function() {
-                    return Function.prototype.apply.apply(method, [obj, arguments]);
-                };
-            }
-        }
-    }
-
-    // Trace() doesn't print the message in IE, so for that case we need to wrap it
-    function traceForIE() {
-        if (console.log) {
-            if (console.log.apply) {
-                console.log.apply(console, arguments);
-            } else {
-                // In old IE, native console methods themselves don't have apply().
-                Function.prototype.apply.apply(console.log, [console, arguments]);
-            }
-        }
-        if (console.trace) console.trace();
-    }
-
     // Build the best logging method possible for this env
     // Wherever possible we want to bind, not wrap, to preserve stack traces
-    function realMethod(methodName) {
-        if (methodName === 'debug') {
-            methodName = 'log';
+    function defaultMethodFactory(methodName, _level, _loggerName) {
+        if (typeof console !== undefinedType) {
+            var consoleMethod = console[methodName] || console.log;
+            if (typeof consoleMethod === "function") {
+                return consoleMethod.bind(console);
+            }
```

Comment:
Nit: imo this'd be clearer if you flip the `if` into a guard condition:

```js
if (typeof console === undefinedType) {
  return noop;
}

...
```

Reply:
So, then we need to return the `noop` in two places. I think you’re basically recommending the logic that I previously had here? 4710d8a5f90f9156869e91a8ff19f3d6115092c8

I was trying to reduce duplication, and this read much clearer to me, but I can put the old logic back.

Reply:
Fair enough. I'd missed the fall-through case there. Personally I think this way round is mildly less clear than the cases, but it's not a big deal! Happy to merge as is, we can tweak and tune endlessly if we want to later on.